numbers/numbers.go

```diff
@@ -157,6 +157,30 @@ func BinomCoefficient(n int, k int) int {
 	return carefulMultDivInt(numer, denom)
 }
 
+//BinomCoefficientLog returns log(n choose k), where log is the natural logarithm. 
+//Ideal for large numbers as this raises the overflow ceiling considerably.
+func BinomCoefficientLog(n int, k int) float64 {
+	if n < 0 || k < 0 || k > n {
+		log.Fatalf("The binomial coefficient call could not be handled: n=%d and k=%d\n", n, k)
+	}
+	if n-k > k {
+		k = n - k
+	}
+	// this special case is handled here so that we don't ask for negative memory for denom
```

Comment:
I think this may not be needed anymore since you are no longer allocating memory